app/scaffold-controllers/welcome-screen/welcomeController.js

```diff
@@ -0,0 +1,135 @@
+
+        this.navigationView = navigationView;
+        this.headerController = headerController;
+    };
+
+    WelcomeController.init = function() {
+        // With a header bar, we use a navigationPlugin to handle
+        // navigation and stacking animations
+        var initWithHeader = function() {
+            return Promise.join(
+                NavigationPlugin.init(),
+                AnchoredLayoutPlugin.init(),
+                WelcomeHeaderController.init(),
+            function(navigationView, layout, headerController) {
+                var loader = navigationView.getLoader();
+                loader.setColor(BaseConfig.loaderColor);
+                navigationView.setHeaderBar(headerController.viewPlugin);
+                headerController.registerBackEventHandler(function() {
+                    navigationView.back();
+                });
+
+                // Remove this line if you wish to enable scrolling
+                // in your welcome screen
+                navigationView.disableScrolling();
+
+                layout.addTopView(headerController.viewPlugin);
+                layout.setContentView(navigationView);
+
+                var welcomeController = new WelcomeController(navigationView, layout, headerController);
+                welcomeController.navigate(WelcomeConfig.url);
+                return welcomeController;
+            });
+        };
+
+        // To navigate without stacking, we use a WebViewPlugin to
+        // handle navigation -- desired behaviour w/o header
+        var initWithoutHeader = function() {
+            return Promise.join(
+                WebViewPlugin.init(),
+                AnchoredLayoutPlugin.init(),
+            function(webView, layout) {
+                var loader = webView.getLoader();
+                loader.setColor(BaseConfig.loaderColor);
+                layout.setContentView(webView);
+
+                // Remove this line to enable scrolling on welcome screen
+                webView.disableScrolling();
+
+                var welcomeController = new WelcomeController(webView, layout);
+                welcomeController.navigate(WelcomeConfig.url);
+
+                return welcomeController;
+            });
+        };
+
+        Astro.registerRpcMethod(AstroRpc.names.welcomeHasHeader, [], function(res) {
+            res.send(null, WelcomeConfig.showHeader);
+        });
+
+        return (WelcomeConfig.showHeader)
+            ? initWithHeader()
+            : initWithoutHeader();
+    };
+
+    WelcomeController.prototype.registerCloseEventHandler = function (callback) {
+        if (!callback) {
+            return;
+        }
+        if (this.headerController) {
+            this.headerController.registerCloseEventHandler(callback);
+        }
+    };
+
+    WelcomeController.prototype.navigate = function(url) {
+        if (!url) {
+            return;
+        }
+
+        var self = this;
+        if (!self.headerController) {
```

Comment:
Ohhh, I get it.  If we don't have a header we just navigate, otherwise we have to do the whole `generateContent()` bit and then navigate. Could you maybe add a comment just to describe what/why you're doing there?  It wasn't obvious to me right off the bat.